File: csharp/MinSwapsToMakeSeqIncreasing.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LeetCode
{
    class MinSwapsToMakeSeqIncreasing
    {
        public int MinSwap(int[] A, int[] B)
        {
            int[] swapDp = new int[A.Length];
            int[] noSwapDp = new int[A.Length];
            swapDp[0] = 1;
            noSwapDp[0] = 0;
            for (int i = 1; i < A.Length; i++)
            {
                swapDp[i] = A.Length;
                noSwapDp[i] = A.Length;
                if (A[i - 1] < A[i] && B[i - 1] < B[i])
                {
                    noSwapDp[i] = noSwapDp[i - 1];
                    swapDp[i] = swapDp[i - 1] + 1;
                }
                if (A[i - 1] < B[i] && B[i - 1] < A[i])
                {
                    noSwapDp[i] = Math.Min(noSwapDp[i], swapDp[i - 1]);
                    swapDp[i] = Math.Min(swapDp[i], noSwapDp[i - 1] + 1);
                }
            }

            return Math.Min(swapDp[A.Length - 1], noSwapDp[A.Length - 1]);
        }
    }
}

File: js/MinSwapsToMakeSeqIncreasing.js
/**
 * @param {number[]} A
 * @param {number[]} B
 * @return {number}
 */
var minSwap = function(A, B) {
    let swap_dp = [];
    let no_swap_dp = [];
    swap_dp[0] = 1;
    no_swap_dp[0] = 0;
    for(let i = 1; i < A.length; i++){
        swap_dp[i] = A.length;
        no_swap_dp[i] = A.length;
        if(A[i-1] < A[i] && B[i-1] < B[i]){
            no_swap_dp[i] = no_swap_dp[i-1];
            swap_dp[i] = swap_dp[i-1] + 1;
        }
        if(A[i-1] < B[i] && B[i-1] < A[i]){
            no_swap_dp[i] = Math.min(no_swap_dp[i], swap_dp[i-1]);
            swap_dp[i] = Math.min(swap_dp[i], no_swap_dp[i-1] + 1);
        }
    }
    return Math.min(swap_dp[A.length - 1], no_swap_dp[A.length - 1]);
};